apps/api/src/user/controller/user.controller.ts

```diff
@@ -143,4 +143,17 @@ export class UserController {
   async createUser(@Body() dto: CreateUserDto) {
     return await this.userService.createUser(dto)
   }
+
+  @Post('validate-otp/:userId')
```

Comment:
- Rename this to `validate-email-change-otp`
- You can get rid of the `:userId` and use `@CurrentUser() user: User` in the controller function parameter since this is a protected method.

Reply:
Added in new commit